infra/resources/_modules/storage_audit/storage.tf

```diff
@@ -0,0 +1,104 @@
+###
+# Immutable Audit Log Storage
+###
+module "immutable_cdc_audit_logs_storage" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v4//storage_account?ref=v7.26.3"
```

Comment:
I have checked that DX module supports all this features: https://github.com/pagopa/dx/blob/main/infra/modules/azure_storage_account/storage_account.tf